common/trieForDirPath.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"strings"
+)
+
+type TrieNode struct {
+	dirPathSeg             map[string]*TrieNode
+	status                 uint32
+	unregisteredButCreated bool
+	isEnd                  bool
+}
+
+type Trie struct {
+	Root *TrieNode
+}
+
+func NewTrie() *Trie {
+	return &Trie{
+		Root: &TrieNode{dirPathSeg: make(map[string]*TrieNode)},
+	}
+}
+
+// @brief InsertStatus inserts the status of the directory path in the trie
+func (t *Trie) InsertStatus(dirPath string, dirCreationStatus uint32) {
```

Comment:
It is also important to note that because this is a map of pointers, we don't have to re-walk the entire tree every time. Any call in this situation, even if we already have the path, is equally expensive time-wise.

We could re-implement the portions of these functions to work on pre-walked nodes, if say, we called GetDirDetails followed by another one of these calls.